#2447 Replace operation name characters which are invalid in code identifiers

joriswit opened the pull request:
A fix for the issue that some api's use characters in operation names that are invalid identifiers, resulting in code that does not compile.
Fix for #2447, #4396

lahma commented on src/NSwag.CodeGeneration/OperationNameGenerators/OperationNameGeneratorHelpers.cs:
```diff
@@ -0,0 +1,62 @@
+        {
+            if (identifier.Length > 0)
+            {
+                var validCategoriesFirstChar = new[] { UnicodeCategory.UppercaseLetter, UnicodeCategory.LowercaseLetter, UnicodeCategory.TitlecaseLetter, UnicodeCategory.ModifierLetter, UnicodeCategory.OtherLetter, UnicodeCategory.LetterNumber };
```
This does not seem ideal. You are allocating arrays on every invocation even though they could be static class fields. Also using `Contains` will use LINQ version instead of more performant array checks.

joriswit replied:
Thanks for your review! I have applied your suggestions.

joriswit commented:
The tests made me realize I have added the invalid character checks in the wrong place. I added it where the current check for the `.` was replaced with the `_`. But this is too early in the process. I have changed it so the check is done much later.

joriswit commented:
Sorry, I had not read the Contributor License Agreement. I don't know if my employer approves this. I'm closing this. Do not merge.
